exercises/practice/saddle-points/spec/saddle_points_spec.cr

```diff
@@ -0,0 +1,58 @@
+require "spec"
+require "../src/*"
+
+describe "SaddlePoints" do
+  it "Can identify single saddle point" do
+    input = [[9, 8, 7], [5, 3, 2], [6, 6, 7]]
+    expected = [{row: 2, column: 1}]
+    SaddlePoints.saddle_points(input).sort_by { |hash| [hash[:row], hash[:column]] }.should eq(expected.sort_by { |hash| [hash[:row], hash[:column]] })
```

Comment:
This is a little scary looking. Could we we move this into a helper so that it's not in every test? Or maybe we could use sets to simplify things:

```suggestion
    SaddlePoints.saddle_points(input).to_set.should eq(expected.to_set)
```

This is nicer to look at but doesn't check the count. We could go a bit further and do:

```suggestion
    actual = SaddlePoints.saddle_points(input)
    actual.size.should eq(expected.size)
    actual.to_set.should eq(expected.to_set)
```

Reply:
Good call!